docs/v1/api.md

```diff
@@ -0,0 +1,239 @@
+# Service Catalog API
```

Comment:
I'd prefer to use the term "Service Controller API" since someone coming in fresh to this might think this is the API exposed by a service broker.  Actually, we should probably have a "terminology" section to make sure we're all using the same terms correctly.